internal/services/compute/disk_encryption_set_data_source.go

```diff
@@ -84,7 +86,16 @@ func dataSourceDiskEncryptionSetRead(d *pluginsdk.ResourceData, meta interface{}
 		d.Set("auto_key_rotation_enabled", props.RotationToLatestKeyVersionEnabled)
 
 		if props.ActiveKey != nil && props.ActiveKey.KeyUrl != "" {
-			d.Set("key_vault_key_url", props.ActiveKey.KeyUrl)
+			keyVaultURI := props.ActiveKey.KeyUrl
+			isHSMURI, err, _, _ := managedHsmHelpers.IsManagedHSMURI(env, keyVaultURI)
```

Comment:
Heads up that this function is being removed in https://github.com/hashicorp/terraform-provider-azurerm/pull/26218 and @wuxu92 is adding a replacement for this in an upcoming PR - so this PR is dependent on https://github.com/hashicorp/terraform-provider-azurerm/pull/26218 / the PR from @wuxu92

Reply:
I am going to approve and merge this to expedite it in due to an internal ask. @wuxu92  can update it in #26218 with the pattern used to remove it